SimPEG/regularization/base.py

```diff
@@ -28,98 +26,266 @@ class BaseRegularization(BaseObjectiveFunction):
 
     """
 
-    def __init__(self, mesh=None, **kwargs):
-        super(BaseRegularization, self).__init__()
-        self.regmesh = RegularizationMesh(mesh)
-        if "indActive" in kwargs.keys():
-            indActive = kwargs.pop("indActive")
-            self.regmesh.indActive = indActive
-        utils.setKwargs(self, **kwargs)
+    _model = None
+
+    def __init__(
+        self,
+        mesh: RegularizationMesh | BaseMesh,
+        active_cells: np.ndarray | None = None,
+        mapping: maps.IdentityMap | None = None,
+        reference_model: np.ndarray | None = None,
+        units: str | None = None,
+        weights: dict | None = None,
+        **kwargs,
+    ):
+        if isinstance(mesh, BaseMesh):
+            mesh = RegularizationMesh(mesh)
+
+        if not isinstance(mesh, RegularizationMesh):
+            raise TypeError(
+                f"'regularization_mesh' must be of type {RegularizationMesh} or {BaseMesh}. "
+                f"Value of type {type(mesh)} provided."
+            )
+
+        self._regularization_mesh = mesh
+        self._weights = {}
 
-    counter = None
+        if active_cells is not None:
+            self.active_cells = active_cells
 
-    # Properties
-    mref = props.Array("reference model")
-    indActive = properties.Array(
-        "indices of active cells in the mesh", dtype=(bool, int)
-    )
-    cell_weights = properties.Array(
-        "regularization weights applied at cell centers", dtype=float
+        self.mapping = mapping
+
+        super().__init__(**kwargs)
+
+        self.reference_model = reference_model
+        self.units = units
+
+        if weights is not None:
+            if not isinstance(weights, dict):
+                weights = {"user_weights": weights}
+            self.set_weights(**weights)
+
+    @property
+    def active_cells(self) -> np.ndarray:
+        """A boolean array of active cells on the regularization
+
+        Returns
+        -------
+        (n_cells, ) Array of bool
+
+        Notes
+        -----
+        If this is set with an array of integers, it interprets it as an array
+        listing the active cell indices.
+        """
+        return self.regularization_mesh.active_cells
+
+    @active_cells.setter
+    def active_cells(self, values: np.ndarray | None):
+        self.regularization_mesh.active_cells = values
+
+        if values is not None:
+            volume_term = "volume" in self._weights
+            self._weights = {}
+            self._W = None
+            if volume_term:
+                self.set_weights(volume=self.regularization_mesh.vol)
+
+    indActive = deprecate_property(
+        active_cells,
+        "indActive",
+        "0.x.0",
+        error=False,
     )
-    regmesh = properties.Instance(
-        "regularization mesh", RegularizationMesh, required=True
+
+    @property
+    def model(self) -> np.ndarray:
+        """Reference physical property model"""
```

Comment:
question here: if I am understanding correctly, this is different from the `reference_model`? but possibly used in the Sparse? (still looking through everything :) 

If so a couple of suggestions / questions
- it would be worth updating the docstring to clarify the difference from the `reference_model` 
- and second, is this something that we want to expose to users? or should it be kept as a private variable under the hood?

Reply:
Replicated docstring - will fix.